htdocs/web_portal/controllers/user/register.php

```diff
@@ -56,10 +56,20 @@ function draw() {
     die();
     }
 
+    //Extract users email from oidc claims
+    $authDetails = $_SERVER['OIDC_CLAIM_external_authn'];
+    $startPos = 3+strpos($authDetails, ":", (strpos($authDetails, "MAIL")));
+    $endPos = strpos($authDetails, "\"", 3+$startPos);
+    $length = $endPos-$startPos;
+    $userEmail = substr($authDetails, $startPos, $length);
+
     /* @var $authToken \org\gocdb\security\authentication\IAuthentication */
     $authToken = Get_User_AuthToken();
     $params['authAttributes'] = $authToken->getDetails();
 
+    $params['given_name'] = $_SERVER['OIDC_CLAIM_given_name'];
```

Comment:
As a note to future Greg, this will need wrapping in a "if using OIDC" statement.

Reply:
or rather everything from `$authDetails = $_SERVER['OIDC_CLAIM_external_authn'];` will need to be.

Reply:
Captured in #325